odl/tomo/geometry/detector.py

```diff
@@ -199,13 +198,15 @@ def __init__(self, part, axis):
         if np.linalg.norm(axis) <= 1e-10:
             raise ValueError('`axis` vector {} too close to zero'
                              ''.format(axis))
-        self._axis = np.asarray(axis) / np.linalg.norm(axis)
-        self._normal = perpendicular_vector(self.axis)
+        self.__axis = np.asarray(axis) / np.linalg.norm(axis)
+        # Let (normal, axis) be positively oriented since the normal will
+        # be the basis for det_to_src.
+        self.__normal = -perpendicular_vector(self.axis)
```

Comment:
Actually, these guys should also have a default configuration of axes and normal, and custom vectors should use a global rotation.